compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -4125,7 +4133,8 @@ private boolean inferFunctionIsolation(BSymbol symbol, IsolationInferenceInfo fu
         }
 
         for (BLangExpression dependsOnArg : functionIsolationInferenceInfo.dependsOnFuncCallArgExprs) {
-            if (!isIsolatedExpression(dependsOnArg)) {
+            if (!isIsolatedExpression(dependsOnArg, false, false, new ArrayList<>(), true,
+                    publiclyExposedObjectTypes, classDefinitions, moduleLevelVariables, new HashSet<>())) {
```

Comment:
Shouldn't we pass `unresolvedSymbols` here? Shall we add a test too?

Reply:
The existing tests will cover this scenario.

Reply:
Updated in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42366/commits/aef9b040ac3a3a01bc0d9108cd0ebaf5bf0405db).

Reply:
But that should have resulted in a stack overflow-ish error, right? If we are recursively checking the same isolated-ness of the same construct?

Reply:
Example:

```ballerina
function fn(any x = ()) returns any {
    _ = start fn(fn());
}
```

Reply:
Thanks!
When we don't pass unresolved symbols, for the provided sample it gives SO.